### PR TITLE
[i2c, dv] Update i2c.hjson to support automation for csr excl

### PR DESCRIPTION
**Pull request by tunghoang290780**
This has been verified with RTL updated (#1703) to leverage #1635 
 
Signed-off-by: Tung Hoang <tung.hoang.290780@gmail.com>

### CODE REVIEW

File: hw/ip/i2c/data/i2c.hjson [hunk 1]
```diff
@@ -70,6 +70,8 @@
       swaccess: "ro"
       hwaccess: "hwo"
       hwext:    "true"
+      tags: [// ro registers - exclude init and write-read check
```
Comment (weicaiyang) [inline hw/ip/i2c/data/i2c.hjson]: nit: comment looks not accurate. only exclude write-read check

File: hw/ip/i2c/data/i2c.hjson [hunk 2]
```diff
@@ -218,6 +222,8 @@
       swaccess: "ro"
       hwaccess: "hwo"
       hwext: "true"
+      tags: [// ro registers - exclude init and write-read check
```
Comment (weicaiyang) [inline hw/ip/i2c/data/i2c.hjson]: ditto

File: hw/ip/i2c/data/i2c.hjson [hunk 3]
```diff
@@ -70,6 +70,8 @@
       swaccess: "ro"
       hwaccess: "hwo"
       hwext:    "true"
+      tags: [// ro registers - exclude init and write-read check
+             "excl:CsrAllTests:CsrExclWriteCheck"]
```
Comment (sriyerg) [inline hw/ip/i2c/data/i2c.hjson]: Make this CsrNonInitTests instead?

File: hw/ip/i2c/data/i2c.hjson [hunk 4]
```diff
@@ -107,6 +109,8 @@
       hwaccess: "hrw"
       hwext: "true"
       hwre: "true"
+      tags: [// ro registers - exclude init and write-read check
```
Comment (sriyerg) [inline hw/ip/i2c/data/i2c.hjson]: Make the comment more meaningful, like: "Reading uninitialized RDATA results in X-propagation; exclude from all checks."
Likewise for others.

File: hw/ip/i2c/data/i2c.hjson [hunk 5]
```diff
@@ -218,6 +222,8 @@
       swaccess: "ro"
       hwaccess: "hwo"
       hwext: "true"
+      tags: [// ro registers - exclude init and write-read check
+             "excl:CsrAllTests:CsrExclWriteCheck"]
```
Comment (sriyerg) [inline hw/ip/i2c/data/i2c.hjson]: CsrAllTests -> CsrNonInitTests

**Comment by sriyerg** [comment]
This updated is needed for chip level CSR tests. I will go ahead and merge it in.

**Pull request merged**